### PR TITLE
IBP-5636-Implement_in-app_feedback_in_manage_program_settings

### PR DESCRIPTION
**Pull request by cuenyad**
Please, review.

Included:
https://github.com/IntegratedBreedingPlatform/BMSAPI/pull/693
https://github.com/IntegratedBreedingPlatform/BMSConfig/pull/114
https://github.com/IntegratedBreedingPlatform/Commons/pull/344
https://github.com/IntegratedBreedingPlatform/Workbench/pull/695

### CODE REVIEW

File: src/main/resources/liquibase/workbench_changelog/20_4_1.xml [hunk 1]
```diff
@@ -21,4 +21,65 @@
...
+		</comment>
+		<sql dbms="mysql" splitStatements="true">
+			UPDATE feedback SET collector_id = '7k444vha', attempts = '5' WHERE (feature = 'GERMPLASM_LIST');
+		</sql>
+	</changeSet>
+
+	<changeSet author="cuenyad" id="v20.4.1-4">
+		<preConditions onFail="MARK_RAN">
+			<and>
+				<tableExists tableName="feedback"></tableExists>
```
Comment (clarysabel) [inline src/main/resources/liquibase/workbench_changelog/20_4_1.xml:69]: @cuenyad I think you can only check the non existance of MANAGE_PROGRAM_SETTINGS in this changeset

File: src/main/resources/liquibase/workbench_changelog/20_4_1.xml [hunk 2]
```diff
@@ -21,4 +21,65 @@
...
+		</sql>
+	</changeSet>
+
+	<changeSet author="cuenyad" id="v20.4.1-3">
+		<preConditions onFail="MARK_RAN">
+			<and>
+				<tableExists tableName="feedback"></tableExists>
+				<columnExists tableName="feedback" columnName="collector_id"/>
+				<columnExists tableName="feedback" columnName="attempts"/>
+				<sqlCheck expectedResult="1">
```
Comment (clarysabel) [inline src/main/resources/liquibase/workbench_changelog/20_4_1.xml:53]: @cuenyad just check that GERMPLASM_LIST exists as a feature

File: src/main/resources/liquibase/workbench_changelog/20_4_1.xml [hunk 3]
```diff
@@ -21,4 +21,65 @@
...
+			</and>
+		</preConditions>
+		<comment>
+			Add collector_id and attempts columns to 'feedback' table
+		</comment>
+		<sql dbms="mysql" splitStatements="true">
+			ALTER TABLE feedback
+				ADD COLUMN collector_id VARCHAR(45) NOT NULL,
+				ADD COLUMN attempts INT(11) NOT NULL DEFAULT 1,
+				CHANGE COLUMN enabled enabled TINYINT(1) NOT NULL DEFAULT '1' AFTER attempts;
```
Comment (clarysabel) [inline src/main/resources/liquibase/workbench_changelog/20_4_1.xml:43]: remove change column @cuenyad

File: src/main/java/org/generationcp/middleware/api/feedback/FeedbackDto.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+	private Integer id;
+
+	private FeedbackFeature feature;
+
+	private String collectorId;
+
+	private Integer attempts;
+
+	private boolean enabled = true;
```
Comment (clarysabel) [inline src/main/java/org/generationcp/middleware/api/feedback/FeedbackDto.java:18]: why do you need to set a default here and why the disable and enable methods are needed? 
@cuenyad

File: src/main/resources/liquibase/workbench_changelog/20_4_1.xml [hunk 5]
```diff
@@ -21,4 +21,54 @@
...
+		<preConditions onFail="MARK_RAN">
+			<sqlCheck expectedResult="0">
+					SELECT count(*) FROM feedback WHERE feature = 'MANAGE_PROGRAM_SETTINGS';
+			</sqlCheck>
+		</preConditions>
+		<comment>
+			Add 'manage program settings' feature to 'feedback' table
+		</comment>
+		<sql dbms="mysql" splitStatements="true">
+			INSERT INTO feedback (feature, collector_id, attempts, enabled) VALUES ('MANAGE_PROGRAM_SETTINGS', 'cfvubhz4', 3, true);
```
Comment (clarysabel) [inline src/main/resources/liquibase/workbench_changelog/20_4_1.xml:70]: not sure if this is 2 or 3. @cuenyad

Comment (cuenyad) [reply src/main/resources/liquibase/workbench_changelog/20_4_1.xml:70]: In the description of the ticket seed. once the user has entered the functionality > 2 times. @clarysabel

**Review by clarysabel — APPROVED** [review]

**Pull request merged**